pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -105,7 +104,6 @@ func NewCloudPrivateIPConfigController(
 				controller.Enqueue(new)
 			}
 		},
-		DeleteFunc: controller.Enqueue,
```

Comment:
> But in context of the bug in question, the deleteFunc triggered here is since the dialer.dial on port 9 for an egress assignable node would become unreachable thereby suggesting we need to activate fail-over mechanism and remove this CloudPrivateIPConfig right?

The `DeleteFunc` for this controller is executed when the object is completely deleted from the API server. This only happens when the finalizer has been removed from the object. The network plugins (ovn-kubernetes and openshift-sdn) will perform client call deletes, but this does not trigger the `DeleteFunc`. Client call deletes will set the `DeletionTimestamp` on the object, but given the presence of the finalizer will only trigger an `UpdateFunc` event for this controller. Once this controller removes the finalizer: we trigger our own `DeleteFunc` since the object finally gets deleted. 

> instead of deleting the handler completely, does it make sense to add some log saying it acknowledges this config going away? 

OK, I can keep the existing code which logs:

```
cloudprivateipconfig_controller.go:405] CloudPrivateIPConfig: "10.0.73.235" in work queue no longer exists
```

That is equivalent to what you just said.